packages/devtools_server/lib/src/server_api.dart

```diff
@@ -148,6 +149,32 @@ class ServerApi {
           request,
           json.encode(_devToolsUsage.surveyShownCount),
         );
+      case apiGetPrimaryAppSizeFile:
```

Comment:
these two cases lookidentical and should probably be merge to a single method that is just about fetching cached files rather than something appSize specific.

Reply:
We also need to avoid serving arbitrary files on the users machine which could lead to security vulnerabilities. serving only files under the `.devtools` directory and prohibiting paths that resolve to a directory outside of .devtools would make this safe.
For example we need to avoid accepting a file path like
`././mypasswords.txt`
as that would resolve to a directory not under `.devtools`

Reply:
App size files may not be in a .devtools directory though. If a user runs the --analyze-size command on their app, the app size files would likely live in their app's directory or wherever else on their machine they choose to specify as output location (for example, when I have run this command, I usually send the files to my Downloads folder).

One thing we could do is change the flutter_tools code to save them not to the flutter app's base directory by default, but instead to a .devtools folder. Right now, we only have a .devtools file where we store client id data, analytics opt-in, survey info, etc., but we could modify this so that we have .devtools folder with the .devtools file inside of it. If we did something like this, would we have the same security concern for accessing files at `<user-home-dir>/.devtools/`?

Reply:
Changed to using ~/.devtools/ directory. Made the change backwards compatible so existing .devtools files will be moved inside of the .devtools/ directory if present